File: 1.py



import socket
blocked_ips = ["1.1.1.1", "8.8.8.8"]
server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server_address = ('', 8000)
server_socket.bind(server_address)
server_socket.listen(1)
print("wait for responce")
while True:
    client_socket, client_address = server_socket.accept()
    print("Received connection from:", client_address)
    if client_address[0] in blocked_ips:
        print("Blocked connection from:", client_address)
        client_socket.close()
        continue
    response = "Hello, client!"
    client_socket.sendall(response.encode())
    client_socket.close()
    
    
    
    

File: 1.rs







use std::net::{TcpListener, TcpStream};
use std::io::{Read, Write};
const BLOCKED_IPS: [&str; 2] = ["1.1.1.1", "8.8.8.8"];
fn main() {
    let listener = TcpListener::bind("127.0.0.1:8000").expect("Failed to bind address");
    println!("waiting for response");
    for stream in listener.incoming() {
        match stream {
            Ok(mut stream) => {
                let client_address = stream.peer_addr().expect("Failed to get client address");
                println!("Received connection from: {:?}", client_address);
                if BLOCKED_IPS.contains(&client_address.ip().to_string().as_str()) {
                    println!("Blocked connection from: {:?}", client_address);
                    continue;
                }
                handle_connection(stream);
            }
            Err(e) => {
                eprintln!("Error: {}", e);
            }
        }
    }
}
fn handle_connection(mut stream: TcpStream) {
    let mut buffer = [0; 1024];
    stream.read(&mut buffer).expect("Failed to read from stream");
    let response = "Hello, client!\n";
    stream.write_all(response.as_bytes()).expect("Failed to write to stream");
    stream.flush().expect("Failed to flush stream");
}



